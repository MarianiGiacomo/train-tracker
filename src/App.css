.vehicle-tracker-header {
    position: fixed;
    z-index: 3000;
    width: 100%;
}

.logo {
    width: 120px;
    height: 31px;
    background: rgba(255, 255, 255, 0.2);
    margin: 16px 24px 16px 0;
    float: left;
}

.vehicle-tracker-menu {
    line-height: 64px;
}

.vehicle-tracker-content-wrapper {
    padding: 0 50px; 
    margin-top: 64px;
}

.vehicle-tracker-content {
    background: #fff;
    padding: 24px;
    min-height: 90vh; 
}

.vehicle-tracker-footer {
    text-align: center; 
}