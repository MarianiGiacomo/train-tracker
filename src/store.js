import {
  createStore, combineReducers, applyMiddleware, compose,
} from 'redux';
import thunk from 'redux-thunk';

import {
  selectedTrainReducer,
  allActiveTrainsReducer,
  trainLocationReducer,
  mqttClientReducer,
  stationMedatadaReducer,
  filteredActiveTrainsReducer,
  messageReducer,
  errorReducer,
} from './reducers';

const reducer = combineReducers({
  trains: allActiveTrainsReducer,
  selectedTrains: selectedTrainReducer,
  trainLocation: trainLocationReducer,
  mqttClient: mqttClientReducer,
  stationsMetadata: stationMedatadaReducer,
  filteredTrains: filteredActiveTrainsReducer,
  message: messageReducer,
  errorMessage: errorReducer,
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const store = createStore(reducer, /* preloadedState, */ composeEnhancers(
  applyMiddleware(thunk),
));

export default store;
