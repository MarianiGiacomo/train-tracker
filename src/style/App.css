
.content-wrapper {
    margin: 10px;
    background: #fff;
}