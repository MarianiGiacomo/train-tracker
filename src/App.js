import React, { Component } from 'react';
import { Layout, Menu } from 'antd';

import VehicleTrackerRest from './components/VehicleTrackerRest';
import VehicleTrackerWss from './components/VehicleTrackerWss';
import VehicleTrackerWsRest from './components/VehicleTrackerWsRest';

import './App.css';

const { Header, Content, Footer } = Layout;

class App extends Component {
  state = {
    selectedMenu: 'wss',
  };

  onMenuItemClick = (e) => {
    this.setState({
      selectedMenu: e.key,
    });
  }

  renderContent = () => {
    const { selectedMenu } = this.state;

    switch (selectedMenu) {
      case 'rest':
        return <VehicleTrackerRest />
      case 'wss': 
        return <VehicleTrackerWss />
      case 'wss-rest':
        return <VehicleTrackerWsRest />
      default: 
        return <VehicleTrackerWss />
    }
  }

  render = () => {
   
    return ( 
     <Layout>
      <Header className="vehicle-tracker-header">
        <div className="logo" />
        <Menu
          theme="dark"
          mode="horizontal"
          defaultSelectedKeys={['wss']}
          className="vehicle-tracker-menu"
          onClick={this.onMenuItemClick}
        >
          <Menu.Item key="wss">WebSocket</Menu.Item>
          <Menu.Item key="rest">REST</Menu.Item>
          <Menu.Item key="wss-rest">WS & REST</Menu.Item>
        </Menu>
      </Header>
      <Content className="vehicle-tracker-content-wrapper">
        <div className="vehicle-tracker-content">
          {this.renderContent()}
        </div>
      </Content>
      <Footer className="vehicle-tracker-footer">
        Vehicle tracker Â©2019
      </Footer>
    </Layout>
    );
  }
}

export default App;