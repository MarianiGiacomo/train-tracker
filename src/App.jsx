import React from 'react';
import { connect } from 'react-redux';

import AppHeader from './components/AppHeader';
import AppContent from './components/AppContent';

import './style/App.css';

import { getAndStoreMqttClient } from './reducers/mqttClientReducer';
import { storeTrainLocation } from './reducers/trainLocationReducer';
import { getAndStoreActiveTrainsByDate } from './reducers/trainsListReducer';
import helperFunctions from './utils/helperFunctions';

const App = (props) => {
  props.getAndStoreActiveTrainsByDate(new Date());
  props.getAndStoreMqttClient((message) => saveCoordinates(message));

  const saveCoordinates = async (message) => {
    const coordinates = helperFunctions.extractTrainLocation(message);
    props.storeTrainLocation(coordinates);
  };

  return (
    <div className="content-wrapper">
      <div className="header-component">
        <AppHeader title="Train Tracker" />
      </div>
      <div className="content-component">
        <AppContent />
      </div>

    </div>
  );
};

const mapDispatchToProps = {
  getAndStoreActiveTrainsByDate,
  getAndStoreMqttClient,
  storeTrainLocation,
};

export default connect(null, mapDispatchToProps)(App);
