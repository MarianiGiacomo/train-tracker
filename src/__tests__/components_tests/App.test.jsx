import React from 'react';
import { Provider } from 'react-redux';
import {
  render,
  cleanup,
} from '@testing-library/react';
import store from '../../store'

import App from '../../App'

afterEach(() => {
  cleanup();
  console.error.mockClear();
  console.warn.mockClear();
});

console.error = jest.fn();
console.warn = jest.fn();

describe('<App />', () => {
  afterEach(cleanup);

  test('Test snapshot', () => {
      const { container } = render(
        <Provider store={store}>
          <App />
        </ Provider>);
      expect(container.firstChild).toMatchSnapshot();
  });

  test('Renders without errors or warnings', () => {
      render(
        <Provider store={store}>
          <App />
        </ Provider>
      );
      expect(console.error).not.toHaveBeenCalled();
      expect(console.warn).not.toHaveBeenCalled();
  });

  test('Page renders the right components', () => {
      const { getByText } = render(
        <Provider store={store}>
          <App />
        </ Provider>
      );
      expect(getByText('Train Tracker')).toBeTruthy();
  });
})


