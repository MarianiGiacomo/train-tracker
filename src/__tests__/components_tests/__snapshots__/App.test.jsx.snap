// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Test snapshot 1`] = `
<div
  class="content-wrapper"
>
  <div
    class="header-component"
  >
    <div>
      <header
        class="header ant-layout-header"
      >
        <h2>
          Train Tracker
        </h2>
      </header>
    </div>
  </div>
  <div
    class="content-component"
  >
    <div>
      <div
        class="modal"
      />
      <div
        class="map-component"
      >
        <div>
          <div
            class="map-element leaflet-container leaflet-grab leaflet-touch-drag"
            style="height: 500px; margin: 1% 20%; position: relative;"
            tabindex="0"
          >
            <div
              class="leaflet-pane leaflet-map-pane"
              style="left: 0px; top: 0px;"
            >
              <div
                class="leaflet-pane leaflet-tile-pane"
              >
                <div
                  class="leaflet-layer titleLayer-element"
                  style="z-index: 1;"
                >
                  <div
                    class="leaflet-tile-container leaflet-zoom-animated"
                    style="left: 0px; top: 0px;"
                  >
                    <img
                      alt=""
                      class="leaflet-tile"
                      role="presentation"
                      src="https://b.tile.openstreetmap.org/15/18654/9484.png"
                      style="width: 256px; height: 256px; left: -45px; top: -173px;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="leaflet-pane leaflet-shadow-pane"
              />
              <div
                class="leaflet-pane leaflet-overlay-pane"
              />
              <div
                class="leaflet-pane leaflet-marker-pane"
              >
                <img
                  alt=""
                  class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                  src="https://upload.wikimedia.org/wikipedia/commons/1/18/Aiga_railtransportation.png"
                  style="margin-left: -15px; margin-top: -20px; width: 30px; height: 40px; left: 0px; top: 0px; z-index: 0;"
                  tabindex="0"
                />
              </div>
              <div
                class="leaflet-pane leaflet-tooltip-pane"
              />
              <div
                class="leaflet-pane leaflet-popup-pane"
              />
            </div>
            <div
              class="leaflet-control-container"
            >
              <div
                class="leaflet-top leaflet-left"
              >
                <div
                  class="leaflet-control-zoom leaflet-bar leaflet-control"
                >
                  <a
                    aria-label="Zoom in"
                    class="leaflet-control-zoom-in"
                    href="#"
                    role="button"
                    title="Zoom in"
                  >
                    +
                  </a>
                  <a
                    aria-label="Zoom out"
                    class="leaflet-control-zoom-out"
                    href="#"
                    role="button"
                    title="Zoom out"
                  >
                    −
                  </a>
                </div>
              </div>
              <div
                class="leaflet-top leaflet-right"
              />
              <div
                class="leaflet-bottom leaflet-left"
              />
              <div
                class="leaflet-bottom leaflet-right"
              >
                <div
                  class="leaflet-control-attribution leaflet-control"
                >
                  <a
                    href="https://leafletjs.com"
                    title="A JS library for interactive maps"
                  >
                    Leaflet
                  </a>
                   | © 
                  <a
                    href="http://osm.org/copyright"
                  >
                    OpenStreetMap
                  </a>
                   contributors
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="search-component"
      >
        <div>
          <div
            class="ant-select-show-search ant-select-auto-complete ant-select ant-select-combobox ant-select-enabled ant-select-allow-clear"
            id="search-station-input"
            style="width: 300px;"
          >
            <div
              aria-autocomplete="list"
              aria-controls="test-uuid"
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-select-selection
            ant-select-selection--single"
              role="combobox"
            >
              <div
                class="ant-select-selection__rendered"
              >
                <div
                  class="ant-select-selection__placeholder"
                  style="display: block; user-select: none;"
                  unselectable="on"
                >
                  Search by station
                </div>
                <ul>
                  <li
                    class="ant-select-search ant-select-search--inline"
                  >
                    <div
                      class="ant-select-search__field__wrap"
                    >
                      <input
                        class="ant-input ant-select-search__field"
                        type="text"
                        value=""
                      />
                      <span
                        class="ant-select-search__field__mirror"
                      >
                        
                         
                      </span>
                    </div>
                  </li>
                </ul>
              </div>
              <span
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <i
                  aria-label="icon: down"
                  class="anticon anticon-down ant-select-arrow-icon"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </i>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="trains-component"
      >
        <div>
          <div
            class="ant-table-wrapper"
          >
            <div
              class="ant-spin-nested-loading"
            >
              <div
                class="ant-spin-container"
              >
                <div
                  class="ant-table ant-table-default ant-table-empty ant-table-scroll-position-left"
                >
                  <div
                    class="ant-table-content"
                  >
                    <div
                      class="ant-table-body"
                    >
                      <table
                        class=""
                      >
                        <colgroup>
                          <col
                            class="ant-table-selection-col"
                          />
                          <col />
                          <col />
                          <col />
                          <col />
                          <col />
                        </colgroup>
                        <thead
                          class="ant-table-thead"
                        >
                          <tr>
                            <th
                              class="ant-table-selection-column"
                            >
                              <span
                                class="ant-table-header-column"
                              >
                                <div>
                                  <span
                                    class="ant-table-column-title"
                                  />
                                  <span
                                    class="ant-table-column-sorter"
                                  />
                                </div>
                              </span>
                            </th>
                            <th
                              class=""
                            >
                              <span
                                class="ant-table-header-column"
                              >
                                <div>
                                  <span
                                    class="ant-table-column-title"
                                  >
                                    Train Numner
                                  </span>
                                  <span
                                    class="ant-table-column-sorter"
                                  />
                                </div>
                              </span>
                            </th>
                            <th
                              class=""
                            >
                              <span
                                class="ant-table-header-column"
                              >
                                <div>
                                  <span
                                    class="ant-table-column-title"
                                  >
                                    Departure Station
                                  </span>
                                  <span
                                    class="ant-table-column-sorter"
                                  />
                                </div>
                              </span>
                            </th>
                            <th
                              class=""
                            >
                              <span
                                class="ant-table-header-column"
                              >
                                <div>
                                  <span
                                    class="ant-table-column-title"
                                  >
                                    Departure Time
                                  </span>
                                  <span
                                    class="ant-table-column-sorter"
                                  />
                                </div>
                              </span>
                            </th>
                            <th
                              class=""
                            >
                              <span
                                class="ant-table-header-column"
                              >
                                <div>
                                  <span
                                    class="ant-table-column-title"
                                  >
                                    Arrival Station
                                  </span>
                                  <span
                                    class="ant-table-column-sorter"
                                  />
                                </div>
                              </span>
                            </th>
                            <th
                              class="ant-table-row-cell-last"
                            >
                              <span
                                class="ant-table-header-column"
                              >
                                <div>
                                  <span
                                    class="ant-table-column-title"
                                  >
                                    Arrival Time
                                  </span>
                                  <span
                                    class="ant-table-column-sorter"
                                  />
                                </div>
                              </span>
                            </th>
                          </tr>
                        </thead>
                        <tbody
                          class="ant-table-tbody"
                        />
                      </table>
                    </div>
                    <div
                      class="ant-table-placeholder"
                    >
                      <div
                        class="ant-empty ant-empty-normal"
                      >
                        <div
                          class="ant-empty-image"
                        >
                          <svg
                            height="41"
                            viewBox="0 0 64 41"
                            width="64"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g
                              fill="none"
                              fill-rule="evenodd"
                              transform="translate(0 1)"
                            >
                              <ellipse
                                cx="32"
                                cy="33"
                                fill="#F5F5F5"
                                rx="32"
                                ry="7"
                              />
                              <g
                                fill-rule="nonzero"
                                stroke="#D9D9D9"
                              >
                                <path
                                  d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                />
                                <path
                                  d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                  fill="#FAFAFA"
                                />
                              </g>
                            </g>
                          </svg>
                        </div>
                        <p
                          class="ant-empty-description"
                        >
                          No Data
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
